test/services/smart-action-field-validator.test.js

```
@@ -50,73 +54,171 @@ describe('services > smart-action-field-validator', () => {
+        expect(() => smartActionFieldValidator.validateField(field, actionName)).toThrow(`type of "${field.field}" on the smart action "${actionName}" must be a valid type. See the documentation for more information. https://docs.forestadmin.com/documentation/reference-guide/fields/create-and-manage-smart-fields#available-field-options`);
       });
     });
   });
+
+  describe('validateFieldChangeHook', () => {
+    it('should not throw if field does not have defined hook', () => {
+      expect.assertions(1);
+
+      const field = { field: 'test' };
+      const actionName = 'actionTest';
+      const hooks = [];
+
+      expect(() => smartActionFieldValidator.validateFieldChangeHook(field, actionName, hooks))
+        .not.toThrow();
+    });
+
+    it('should not throw if field have defined hook', () => {
+      expect.assertions(1);
+
+      const field = { field: 'test', hook: 'onChange' };
+      const actionName = 'actionTest';
+      const hooks = {
+        onChange: () => {},
+      };
+
+      expect(() => smartActionFieldValidator.validateFieldChangeHook(field, actionName, hooks))
+        .not.toThrow();
+    });
+
+    it('should throw if field have not defined hook', () => {
+      expect.assertions(1);
+
+      const field = { field: 'test', hook: 'onChange' };
+      const actionName = 'actionTest';
+      const hooks = {};
+
+      expect(() => smartActionFieldValidator.validateFieldChangeHook(field, actionName, hooks))
+        .toThrow(`The hook "${field.hook}" of "${field.field}" field on the smart action "${actionName}" is not defined.`);
+    });
+  });
+
+  describe('validateSmartActionFields', () => {
+    it('should do nothing if action does not have field', () => {
```

Same here

`when the action does not have any fields`.`it should do nothing`
`when the action uses incorrect parameters`.`it should throw an error for fields not being an array` 
...

---

😍